wire-runtime-swift/src/main/swift/ProtoCodable/ProtoEncoder.swift

```diff
@@ -82,13 +82,40 @@ public final class ProtoEncoder {
 
         let writer = ProtoWriter(
             data: .init(capacity: structSize),
-            outputFormatting: [],
+            outputFormatting: outputFormatting,
             rootMessageProtoSyntax: T.self.protoSyntax ?? .proto2
         )
-        writer.outputFormatting = outputFormatting
+
         try value.encode(to: writer)
 
         return Data(writer.buffer, copyBytes: false)
     }
 
+    public func encodeSizeDelimited<T: ProtoEncodable>(_ values: [T]) throws -> Data {
+        // Use the size of the struct as an initial estimate for the space needed.
+        let structSize = MemoryLayout.size(ofValue: T.self)
+
+        let fullBuffer = WriteBuffer(capacity: structSize * values.count)
```

Comment:
Should this calculation attempt to reserve some capacity for the varints or just let the buffer expand naturally?

Reply:
Updated to reserve space 👍